pandas/core/ops.py

```diff
@@ -1629,16 +1663,38 @@ def f(self, other, axis=0):
     return f
 
 
-panel_special_funcs = dict(arith_method=_arith_method_PANEL,
-                           comp_method=_comp_method_PANEL,
-                           bool_method=_arith_method_PANEL)
-
-panel_flex_funcs = dict(flex_arith_method=_flex_method_PANEL,
-                        flex_comp_method=_comp_method_PANEL)
-
 # -----------------------------------------------------------------------------
 # Sparse
 
+def _cast_sparse_series_op(left, right, opname):
+    """
+    For SparseSeries operation, coerce to float64 if the result is expected
+    to have NaN or inf values
+
+    Parameters
+    ----------
+    left : SparseArray
+    right : SparseArray
+    opname : str
+
+    Returns
+    -------
+    left : SparseArray
+    right : SparseArray
+    """
+    opname = opname.strip('_')
+
+    if is_integer_dtype(left) and is_integer_dtype(right):
+        # series coerces to float64 if result should have NaN/inf
+        if opname in ('floordiv', 'mod') and (right.values == 0).any():
```

Comment:
this should go thru the fill_zeros path.

Reply:
Is this obvious?  I don't know the sparse code all that well, have been assuming that the do-it-upfront approach taken here was for a reason.

Reply:
There will need to be a pass soon to make Series division handle division by zero the way that Index division now does.  That might be a reasonable time to take a close look at this.

Reply:
ok pls add it to the list.